sklearn/linear_model/_stochastic_gradient.py

```diff
@@ -1125,7 +1125,7 @@ class SGDClassifier(BaseSGDClassifier):
         an int greater than 1, averaging will begin once the total number of
         samples seen reaches `average`. So ``average=10`` will begin
         averaging after seeing 10 samples.
-        Integer values must be in the range `[1, n_samples]`.
+        Integer values must be in the range `[0, n_samples]`.
```

Comment:
I agree the parameter constraint is `[0, n_samples]`, but I don't understand why. Is using `0` just a different way of saying `average=True`, as in "start right away"? If yes that is surprising because normally I'd think of `0`, in the context of `bool` as `False`. So my question is, should we update the constraint to match the doc string (in the text it also says "if set to a value greater than 1")?

Maybe someone with more knowledge can comment.

Reply:
Looking at the code around, I would correct the parameter validation here because we have a lot of code with something like `if self.average > 0`.

Reply:
@betatim @glemaitre Would you like me to revert the commit that updates the docstrings relative to the `average` parameter (it's isolated in https://github.com/scikit-learn/scikit-learn/pull/28373/commits/d181bda35cd5e7dcca9bce3366052625760fc14e) and open an issue regarding `average` constraints and usage?

Reply:
Indeed this would be easier to handle in a separate PR.

Reply:
@betatim @glemaitre I opened an issue for this [here](https://github.com/scikit-learn/scikit-learn/issues/28389). Feel free to edit it as you see fit.